app/uk/gov/hmrc/exports/connectors/CustomsDataStoreConnector.scala

```diff
@@ -16,23 +16,27 @@
 
 package uk.gov.hmrc.exports.connectors
 
-import javax.inject.Inject
-
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
+import play.api.Logging
 import play.api.http.Status.NOT_FOUND
 import uk.gov.hmrc.exports.config.AppConfig
 import uk.gov.hmrc.exports.models.emails.VerifiedEmailAddress
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, Upstream4xxResponse}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, Upstream4xxResponse, UpstreamErrorResponse}
 import uk.gov.hmrc.http.HttpReads.Implicits._
 
-class CustomsDataStoreConnector @Inject()(http: HttpClient)(implicit appConfig: AppConfig, ec: ExecutionContext) {
+@Singleton
+class CustomsDataStoreConnector @Inject()(http: HttpClient)(implicit appConfig: AppConfig, ec: ExecutionContext) extends Logging {
 
   import CustomsDataStoreConnector._
 
   def getEmailAddress(eori: String)(implicit hc: HeaderCarrier): Future[Option[VerifiedEmailAddress]] =
-    http.GET[Option[VerifiedEmailAddress]](verifiedEmailUrl(eori)).recover {
-      case Upstream4xxResponse(_, NOT_FOUND, _, _) => None
+    http.GET[Option[VerifiedEmailAddress]](verifiedEmailUrl(eori)).recoverWith {
+      case Upstream4xxResponse(_, NOT_FOUND, _, _) => Future.successful(None)
```

Comment:
I don't think this exception will ever be thrown for a 404 (just a None returned from the GET call) as per: [Tech Space Blog](https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=TEC&postingDay=2020%2F6%2F30&title=Http-verbs+update)